casper/src/main/scala/coop/rchain/casper/merging/DeployChainIndex.scala

```diff
@@ -17,7 +19,16 @@ final case class DeployChainIndex(
     postStateHash: Blake2b256Hash,
     eventLogIndex: EventLogIndex,
     stateChanges: StateChange
-)
+) {
+  // equals and hash overrides are required to make conflict resolution faster, particularly rejection options calculation
+  override def equals(obj: Any): Boolean = obj match {
+    case that: DeployChainIndex => that.deploysWithCost.map(_.id) == this.deploysWithCost.map(_.id)
+    case _                      => false
+  }
+
+  override def hashCode(): Int =
+    deploysWithCost.map(_.id).foldLeft(0)((acc, v) => acc + v.hashCode() * 31)
```

Comment:
why `31` here?

Reply:
Its just a prime number to get (used in other places to override hashCode). https://www.artima.com/articles/how-to-write-an-equality-method-in-java this is useful reference. I'm not sure this is the best implementation for hashCode for this case, just did as in other parts of the code. The goal here is to get good distribution of hash codes, so operations on Sets of these data is fast, as hash code defines bucket and then equals resolves collisions.

Reply:
There is helper function to build hash code from arrays. It's plain for loop so maybe little faster.
https://www.baeldung.com/java-hashcode#standard-hashcode-implementations

Reply:
I've tried several implementation. For some reason behaviour on live network and in synthetic test in this PR is different. E.g. if I remove benefit from hashCode completely, making it fixed number, test is much faster
```
override def hashCode(): Int = 0

5 => 2.980869 ms vs 585.726633 ms 
10 => 19.294805 ms vs 13.1713 ms 
15 => 37.885004 ms vs 14.579285 ms 
20 => 37.878952 ms vs 17.656214 ms 
25 => 126.036481 ms vs 58.800682 ms 
30 => 163.905676 ms vs 56.152606 ms 
35 => 329.156289 ms vs 68.314901 ms 
40 => 654.245934 ms vs 97.642741 ms 
45 => 1.58764128 sec vs 158.560366 ms 
50 => 3.521030067 sec vs 210.773154 ms 
```
but in real network slower
```
rejection options (215) in 1.11669206875 min 
rejection options (287) in 2.31602600755 min

```
This is the reference as per current code
```
// Current version
override def hashCode(): Int = deploysWithCost.map(_.id).foldLeft(0)((acc, v) => acc + v.hashCode() * 31)

2.033679 ms vs 605.306271 ms 
12.59387 ms vs 45.038413 ms 
58.595065 ms vs 42.617381 ms 
74.587689 ms vs 152.469986 ms 
46.475547 ms vs 695.93373 ms 
198.669401 ms vs 612.737869 ms 
573.537734 ms vs 2.038174953 sec 
897.53519 ms vs 8.299223488 sec 
3.634718809 sec vs 9.800560232 sec 
7.892337706 sec vs 19.930452911 sec 

rejection options (265) in 3.618607435 sec
rejection options (252) in 1.413084469 sec
```

We might want to optimise it in future, for now I just use helper as @tgrospic suggested.